app/controllers/posts_controller.rb

```diff
@@ -584,14 +584,55 @@ def feature
     render json: { status: 'success', success: true }
   end
 
+  # saving by-field is kept for backwards compatibility with old drafts
   def save_draft
-    key = "saved_post.#{current_user.id}.#{params[:path]}"
+    expiration_time = 86_400 * 7
+
+    base_key = "saved_post.#{current_user.id}.#{params[:path]}"
+    RequestContext.redis.set base_key, params[:body]
+    RequestContext.redis.expire base_key, expiration_time
+
     saved_at = "saved_post_at.#{current_user.id}.#{params[:path]}"
-    RequestContext.redis.set key, params[:post]
     RequestContext.redis.set saved_at, DateTime.now.iso8601
-    RequestContext.redis.expire key, 86_400 * 7
-    RequestContext.redis.expire saved_at, 86_400 * 7
-    render json: { status: 'success', success: true, key: key }
+    RequestContext.redis.expire saved_at, expiration_time
+
+    if params.key?(:comment)
```

Comment:
Feels like this set of `if`s could be made into a loop over an array of the field names. Something like:

```ruby
[:comment, :excerpt, :license, :tag_name, :tags, :title].each do |key|
  if params.key? key
    key_name = "saved_post.#{current_user.id}.#{params[:path]}.#{key.to_s}
    RequestContext.redis.set key_name, params[key]
    RequestContext.redis.expire key_name, expiration_time
  end
end
```

Reply:
Yeah, agreed